### ISSUE TITLE
FNX-12049 ⁃ [Bug] In Firefox Nightly all Sync items are enabled even though I've never enabled them.

### ISSUE THREAD

**Issue by ghost** [mozilla-mobile/fenix#11172]
* [x] AC bug: https://github.com/mozilla-mobile/android-components/issues/7233
* [x] AS bug: https://github.com/mozilla/application-services/issues/3205

## Steps to reproduce
In Firefox Nightly all Sync items are enabled. Even though I have never enabled them. I have only enabled Bookmarks sync option in `77.0.0-beta.1`.

### Expected behavior
Only items that were previously enabled should be enabled

### Actual behavior
All Sync items are enabled automatically

### Device information
* Android device: Motorola One Power, Android 10.
* Fenix version: Nightly 200603 06:00
![Screenshot](https://user-images.githubusercontent.com/42088907/83606669-87fee700-a569-11ea-9641-e40e9cc6431e.png)UNITO-UNDERSCORE!20200603-123329!

**Comment by liuche** [comment]
Hi @SS1113 do you have Synced Tabs enabled on your other accounts? Currently, any of these items are global settings, so it should take your setting from any of your Firefox Account devices.

**Comment by SS1113** [comment]
@liuche No, My other account has only Bookmarks sync enabled and other options are disabled.

**Comment by liuche** [comment]
@Vishwa-Mozilla can you take a look at this bug? We want to match the sync state of the Firefox Account linked, so make sure that it's not "turned on by default".

**Comment by Vishwa-Mozilla** [comment]
I can reproduce this issue but it only happens when a new device is connected until there is a new change in sync settings from the server - doSync() function in WorkManagerSyncManager (A-C) handles fetching any changes to declined/accepted SyncEngines and they did not contain the correct information

**Comment by Vishwa-Mozilla** [comment]
The data that is inside the declined/accepted SyncEngines is then passed onto SyncEnginesStorage which stores the information about SyncEngines with a boolean in SharedPreferences -> this is used to determine the checked status of the sync items

**Comment by Vishwa-Mozilla** [comment]
Create A-C ticket -  https://github.com/mozilla-mobile/android-components/issues/7233

**Comment by SS1113** [comment]
@Vishwa-Mozilla @liuche I forgot to mention that I installed the `Nightly 200603 06:00` freshly and then connected my Firefox account. 

As for the devices, I used the same device while adding the account but I had added the account on different builds.

Firefox Account I had connected to before adding it in Nightly: 

• [Firefox for Android](https://play.google.com/store/apps/details?id=org.mozilla.firefox)

• Firefox Beta 77.0.0-beta.1

**Comment by liuche** [comment]
Aim to get this in 5.3

**Comment by ekager** [comment]
A fix in AS has landed, and AC has been updated today to include that new version in https://github.com/mozilla-mobile/android-components/pull/7423 

I just updated Fenix to get the new AC version in #11707 so let's test in Nightly!

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest Nightly build from 6/19 with Google Pixel (Android 10), Sony Xperia Z5 Premium (Android 7.1.1), and Windows 10.

* I've disabled the syncing of History and Logins, updated the build, and those options remained disabled. I've signed in with another device, synced, and those options remained disabled.
* I've disabled the syncing of Bookmarks, Tabs and Logins, updated the build, and those options remained disabled, only History remained enabled. I've signed in with another device, synced, and those options remained disabled.
